seacatauth/generic.py

```diff
@@ -137,6 +137,34 @@ async def nginx_introspection(
 	return response
 
 
+def parse_url(url: str):
+	"""
+	Parse the URL into a dictionary
+	"""
+	return urllib.parse.urlparse(url)._asdict()
+
+
+def unparse_url(
+	scheme: str = "",
+	netloc: str = "",
+	path: str = "",
+	params: str = "",
+	query: str = "",
+	fragment: str = ""
+):
+	"""
+	Build URL from individual components
+
+	Example usage:
+	```python
+	parsed = parse_url("http://local.test?option=true")
+	parsed["path"] = "/some/subpath"
+	url = unparse_url(**parsed)
+	```
+	"""
+	return urllib.parse.urlunparse((scheme, netloc, path, params, query, fragment))
```

Comment:
These are kind of "null functions".
Do we need them here?

Reply:
Not sure what you mean by null function. I just find the `ParseResult` object from urllib too quirky to work with because it is immutable. And calling the `urlunparse` is always too verbose. 

I simply want to parse url, edit its components and unparse it again.